packages/discovery-react-components/src/components/DocumentPreview/components/PdfHighlight/utils/textBoxMapping/TextProvider.ts

```diff
@@ -50,14 +51,30 @@ export class TextProvider {
 
   /**
    * Get how the given `text` matches to the currently available text
+   * @param text text to search
+   * @param { minLength, maxLength, searchSpan } options match options
+   *   minLength, maxLength: specify min/max length of the match.
+   *   searchSpan: the span where the `text` is searched
    */
-  getMatches(text: string, minLength = 1, maxLength = text.length): TextMatch[] {
+  getMatches(
+    text: string,
+    {
+      minLength = 1,
+      maxLength = text.length,
+      searchSpan
+    }: { minLength?: number; maxLength?: number; searchSpan?: TextSpan } = {}
```

Comment:
Instead of having a positional arg and then an "options" object, why not just make a single arguments object with everything:
```suggestion
    { 
      text,
      minLength = 1,
      maxLength = text.length,
      searchSpan
    }: { 
      text: string;
      minLength?: number;
      maxLength?: number;
      searchSpan?: TextSpan
    } = {}
```

Here and other implementations of `getMatches()`

Reply:
Taking `text` argument is because it's more important than others. Others are options.